src/tensor_ops/realize_to.rs

```diff
@@ -0,0 +1,213 @@
+use crate::{shapes::*, tensor::*};
+
+/// Changes order of dimensions/axes
+pub trait RealizeTo: HasErr + HasShape {
+    /// Realizes the concrete shape of the tensor as another compatable shape:
+    /// ```rust
+    /// # use dfdx::prelude::*;
+    /// # let dev: Cpu = Default::default();
+    /// let a: Tensor<Rank2<2, 3>, f32, _> = dev.zeros();
+    /// let a = a.realize::<(usize, usize)>().unwrap();
+    /// let a = a.realize::<Rank2<2, 3>>().unwrap();
+    /// ```
+    fn realize<Dst: Shape>(self) -> Option<Self::WithShape<Dst>>
+    where
+        Self::Shape: RealizeShapeTo<Dst>,
+    {
+        self.try_realize().unwrap()
+    }
+
+    /// Fallible version of [RealizeTo::realize].
+    fn try_realize<Dst: Shape>(self) -> Result<Option<Self::WithShape<Dst>>, Self::Err>
+    where
+        Self::Shape: RealizeShapeTo<Dst>;
+}
+
+impl<S: Shape, E: Dtype, D: DeviceStorage, T: Tape<E, D>> RealizeTo for Tensor<S, E, D, T> {
+    fn try_realize<Dst: Shape>(self) -> Result<Option<Self::WithShape<Dst>>, Self::Err>
+    where
+        Self::Shape: RealizeShapeTo<Dst>,
+    {
+        Ok(self.shape.realized().map(|dst_shape| Tensor {
+            id: self.id,
+            data: self.data,
+            strides: dst_shape.strides(),
```

Comment:
Also, this needs to have the same value as self.strides, because `dst_shape.strides()` is not necessarily the same as the input strides when the input is broadcasted/permuted.

Reply:
Yep agree, this can just be
```suggestion
            strides: self.strides,
```